core/cont/src/TList.cxx

```diff
@@ -538,17 +538,20 @@ void TList::DeleteLink(TObjLink *lnk)
 
 TObject *TList::FindObject(const char *name) const
 {
+   if (!name)
```

Comment:
I understand your argument, but technically, the thread is not really accessing the data from TList before the `R__COLLECTION_READ_GUARD`, so this change does not affect your ability to trace bugs. I made the change thinking about the possibility that in the future this will not be a noop.

Reply:
However, TList is known to not be thread safe and using it from several threads (without an external lock) is always a mistake no matter how far it goes.  Moving the guard to later in the function actually only reduce the effectiveness of the collision detection.

I also have no intention of making the class named TList to be thread safe (we will always need a container that does not pay the heavy price of locking for the many case where the container will not be shared).  

There might be a derived (and/or sibling) class name TListWIthLock or TListWIthRWLock or any better name.

>  I made the change thinking about the possibility that in the future this will not be a noop.

Not sure what you meant (I suspect you meant 'when it becomes a lock) since it is not a noop if you set the #define the macro R__CHECK_COLLECTION_MULTI_ACCESS.